GitCommands/Git/Executable.cs

```diff
@@ -150,8 +150,11 @@ private void OnProcessExit(object sender, EventArgs eventArgs)
                             if (_throwOnErrorExit && exitCode != 0)
                             {
                                 string errorOutput = _process.StandardError.ReadToEnd().Trim();
-                                ExternalOperationException ex
-                                    = new(command: _process.StartInfo.FileName,
+
+                                Exception ex
+                                    = GitDubiousOwnershipSecurityException.IsGitDubiousOwnershipSecurityError(errorOutput)
+                                ? new GitDubiousOwnershipSecurityException(errorOutput)
```

Comment:
This isn't good - this adds a hard dependency between different projects to implementation details. It doesn't feel to do this at this layer.
We're passing `errorOutput` verbatim wrapped in an inner exception, so at the UI layer - where we present the exception - we should be able to inspect it and provide different UI/handling.